Tests/ContentfulTests/QueryTests.swift

```diff
@@ -512,12 +545,17 @@ class QueryTests: XCTestCase {
 
         let query = QueryOn<Dog>(where: "fields.description", .matches("bacon pancakes"))
 
-        QueryTests.client.fetchMappedEntries(with: query).then { dogsResponse in
-            let dogs = dogsResponse.items
-            expect(dogs.count).to(equal(1))
-            expect(dogs.first?.name).to(equal("Jake"))
+        QueryTests.client.fetchMappedEntries(with: query) { result in
+            switch result {
+            case .success(let dogsResponse):
+                let dogs = dogsResponse.items
+                expect(dogs.count).to(equal(1))
+                expect(dogs.first?.name).to(equal("Jake"))
+            case .error(let error):
+                fail("Should not throw an error \(error)")
```

Comment:
does failing fullfill the expectation too? Or would it wait for ever in here now?

Reply:
No, the expectation is fulfilled outside of the switch: no matter the case, it is fulfilled.